distributed/comm/ucx.py

```diff
@@ -221,13 +219,13 @@ async def write(
 
                 # Send frames
 
-                # It is necessary to first synchronize the default stream before start sending
-                # We synchronize the default stream because UCX is not stream-ordered and
-                #  syncing the default stream will wait for other non-blocking CUDA streams.
+                # It is necessary to first synchronize the per-thread default stream before start sending
+                # We synchronize the per-thread default stream because UCX is not stream-ordered and
+                #  syncing the per-thread default stream will wait for other non-blocking CUDA streams.
```

Comment:
It won't wait for other threads' default per-thread default streams...

Reply:
So should we leave this as-is? Or is there something else we should be doing here?

Reply:
I think Mark is referring to the section of `syncing the per-thread default stream will wait for other non-blocking CUDA streams` as that line is incorrect. Syncing the per-thread default stream will only wait for the current stream WITHOUT blocking work on other non-blocking CUDA streams.

Reply:
Right, I just wasn't sure if this implied changes beyond the text in the comment.

Reply:
Assuming syncing only the PTDS is correct behaviour, I think only the comment is incorrect.